homeassistant/components/guardian/util.py

```diff
@@ -50,3 +62,26 @@ async def _async_update_data(self) -> dict[str, Any]:
             except GuardianError as err:
                 raise UpdateFailed(err) from err
         return cast(dict[str, Any], resp["data"])
+
+    async def async_initialize(self) -> None:
+        """Initialize the coordinator."""
+
+        @callback
+        def async_reboot_requested() -> None:
+            """Respond to a reboot request."""
+            self.last_update_success = False
+            self.async_update_listeners()
+
+        self._signal_handler_unsubs.append(
+            async_dispatcher_connect(
+                self.hass, self.signal_reboot_requested, async_reboot_requested
+            )
+        )
+
+        @callback
+        def async_teardown(_: Event) -> None:
+            """Tear the coordinator down appropriately."""
+            for unsub in self._signal_handler_unsubs:
+                unsub()
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_teardown)
```

Comment:
Is there an opportunity to use `entry.async_on_unload` for these?

Reply:
Great call! 👍🏻